app/assets/scss/overrides/_govuk-label.scss

```diff
@@ -13,3 +13,8 @@
 
   margin-bottom: govuk-spacing(2);
 }
+
+.govuk-label--l {
```

Comment:
Yup, the `.govuk-label` override takes precedence over previous styling, so is ignoring the `.govuk-label--l` styles before it.